### PR TITLE
Fix typos and spacing issues in contributing, security, changelog, issue templates, and \eng\ markdowns.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Follow up to the PR https://github.com/Azure/azure-sdk-for-c/pull/959, related to https://github.com/Azure/azure-sdk-for-c/issues/459

Also fixed some typos in API doc comments and code.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <!-- BEGIN MICROSOFT SECURITY.MD V0.0.5 BLOCK -->
 
-## Security
```
Comment (ahsonkhan) [inline SECURITY.md:3]: Note to self: Revert this and fix it in the appropriate place.

File: eng/common/README.md [hunk 2]
```diff
@@ -4,9 +4,9 @@ The `eng/common` directory contains engineering files that are common across the
 It should remain relatively small and only contain textual based files like scripts, configs, or templates. It
 should not contain binary files as they don't play well with git.
 
-# Updating
```
Comment (ahsonkhan) [inline eng/common/README.md:7]: Note to self: Once merged, open a PR in azure-sdk-tools with the changes to eng/common.

File: eng/docs/index/templates/matthews/partials/class.tmpl.partial [hunk 3]
```diff
@@ -26,7 +26,7 @@
 <div id="collapsible-{{id}}" class="collapse in">
 <div class="markdown level1 summary">{{{summary}}}</div>
 <div class="markdown level1 conceptual">{{{conceptual}}}</div>
-<h5 class="decalaration">{{__global.declaration}}</h5>
```
Comment (ahsonkhan) [inline eng/docs/index/templates/matthews/partials/class.tmpl.partial:29]: @danieljurek should I revert this change and wait for [this](https://github.com/MathewSachin/docfx-tmpl/pull/4) to get merged instead or is it OK to fix this here?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -26,15 +26,17 @@ A clear and concise description of what you expected to happen.
 If applicable, add screenshots to help explain your problem.
 
 **Setup (please complete the following information):**
- - OS: [e.g. iOS]
- - IDE : [e.g. IntelliJ]
- - Version of the Library used
+
+- OS: [e.g. iOS]
+- IDE : [e.g. IntelliJ]
```
Comment (antkmsft) [inline .github/ISSUE_TEMPLATE/bug_report.md:31]:
```suggestion
- IDE: [e.g. IntelliJ]
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -26,15 +26,17 @@ A clear and concise description of what you expected to happen.
...
+- Version of the Library used
 
 **Additional context**
 Add any other context about the problem here.
 
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
```
Comment (antkmsft) [inline .github/ISSUE_TEMPLATE/bug_report.md:38]: Consider changing it to "please" - https://ell.stackexchange.com/questions/51618/difference-between-kindly-and-please

File: .github/ISSUE_TEMPLATE/question-query-template.md [hunk 6]
```diff
@@ -14,11 +14,13 @@ A clear and concise ask/query.
 A clear explanation of why is this not a bug or a feature request?
 
 **Setup (please complete the following information if applicable):**
- - OS: [e.g. iOS]
- - IDE : [e.g. IntelliJ]
- - Version of the Library used
- 
+
+- OS: [e.g. iOS]
+- IDE : [e.g. IntelliJ]
```
Comment (antkmsft) [inline .github/ISSUE_TEMPLATE/question-query-template.md:19]:
```suggestion
- IDE: [e.g. IntelliJ]
```

File: eng/common/TestResources/New-TestResources.ps1.md [hunk 7]
```diff
@@ -283,7 +299,7 @@ created resource group.
 An optional cleanup process can delete resource groups whose "DeleteAfter"
 timestamp is less than the current time.
 
-This isused for CI automation.
+This issued for CI automation.
```
Comment (antkmsft) [inline eng/common/TestResources/New-TestResources.ps1.md:302]: I think this meant to be "is used", not "issued" :)

File: eng/docs/index/templates/matthews/partials/class.tmpl.partial (hunk 3 above)
Comment (danieljurek) [reply eng/docs/index/templates/matthews/partials/class.tmpl.partial:29]: No problem with merging this right now as we're not doing doc index generation today. I have added this to the list of things to do in this issue: https://github.com/Azure/azure-sdk-for-c/issues/930

File: eng/common/README.md (hunk 2 above)
Comment (weshaggard) [reply eng/common/README.md:7]: What caused this to be a H2 instead of H1?

Comment (ahsonkhan) [reply eng/common/README.md:7]: [MD025](https://github.com/DavidAnson/markdownlint/blob/v0.20.4/doc/Rules.md#md025) - Multiple top level headings in the same document

Comment (weshaggard) [reply eng/common/README.md:7]: Thanks. Would you mind also filing another issue in azure-sdk repo to consider using this MD linter for all the language repos?

Comment (ahsonkhan) [reply eng/common/README.md:7]: > Note to self: Once merged, open a PR in azure-sdk-tools with the changes to eng/common.

Done in https://github.com/Azure/azure-sdk-tools/pull/892 by @chidozieononiwu. Thanks!

Comment (ahsonkhan) [reply eng/common/README.md:7]: > Thanks. Would you mind also filing another issue in azure-sdk repo to consider using this MD linter for all the language repos?

Done. See https://github.com/Azure/azure-sdk/issues/1698
